cranelift/codegen/src/opts/algebraic.isle

```diff
@@ -240,3 +240,84 @@
 (rule (simplify
        (icmp (ty_int ty) (IntCC.SignedLessThanOrEqual) x x))
       (iconst ty (imm64 1)))
+
+;; Generic icmp reversal rule.
+(rule (simplify (icmp ty cc x y)) (icmp ty (intcc_reverse cc) y x))
```

Comment:
@jameysharp and I talked a little further about this today and we agreed that we probably don't want to double the number of `icmp` nodes unconditionally; it's cheaper to have more rules, some with inverted matching forms on the left-hand sides, than it is to grow the egraph and then have to match on the result of that. Would you be willing to remove this rule and compose it into the below manually instead?